exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/efgs/CallbackInitializer.java

```diff
@@ -0,0 +1,83 @@
+package fi.thl.covid19.exposurenotification.efgs;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+import static net.logstash.logback.argument.StructuredArguments.keyValue;
+
+@Component
+@Profile("!dev & !test")
```

Comment:
While this is fine, it's generally better that profiles configure features than that features know what profiles they associate with. Here, it would mean binding this to some callback-enabled property (there was going to be one anyhow, right?) and set that to false in dev- and test- application.yml:s. That way, the dependency direction is flipped and the pretty implementation class doesn't have to know about possible runtime-environments.

Reply:
Yep, but in this case we then would need two enabled switches, one which enables whole bean and one which enables callback. It seemed kind of messy. Anyway, this is matter of opinion, I might chańge it in future versions.